contracts/multisig-prover/src/encoding/mod.rs

```diff
@@ -1,10 +1,92 @@
 pub mod abi;
+pub mod bcs;
 
+use axelar_wasm_std::hash::Hash;
 use cosmwasm_schema::cw_serde;
+use cosmwasm_std::HexBinary;
+use error_stack::Result;
+use multisig::key::Signature;
+use multisig::msg::SignerWithSig;
+use multisig::verifier_set::VerifierSet;
+
+use crate::error::ContractError;
+use crate::payload::Payload;
 
 #[cw_serde]
 #[derive(Copy)]
 pub enum Encoder {
     Abi,
     Bcs,
 }
+
+impl Encoder {
```

Comment:
I think we should create an `Encoder` trait instead that AbiEncoder and BcsEncoder types implement.
`EncoderType` enum can be used to denote which encoders are supported, and returns the appropriate encoders. Avoids having to match cases for each method below then

Reply:
Not sure if that is better and I also want to avoid making this PR more complex and bigger for some refactor that we can do later.

Reply:
Ok, worth doing it sooner rather than later still since we'll be adding Stellar as well